lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2022 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+class PollBasedFilePoller extends AbstractFilePoller {
+  private final AtomicBoolean changedSinceLastPoll = new AtomicBoolean(false);
+  private volatile long lastUpdatedAt = 0;
+
+  protected PollBasedFilePoller(Path filePath, Duration pollInterval) {
+    super(filePath);
+    if (pollInterval == null) {
+      throw new IllegalArgumentException("Poll interval cannot be null");
+    }
+
+    ScheduledExecutorService scheduledExecutorService =
+        Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactory() {
+              @Override
+              public Thread newThread(Runnable r) {
+                Thread t = new Thread(null, r, "PollBasedFilePoller", 0);
+                t.setDaemon(true);
+                return t;
+              }
+            });
+    scheduledExecutorService.scheduleAtFixedRate(
+        this::poll, pollInterval.toNanos(), pollInterval.toNanos(), TimeUnit.NANOSECONDS);
```

Comment:
Not sure if this is desired or what are the repercussions, but you could pass `0` as `initialDelay` in  `scheduleAtFixedRate` and it would run the first poll right away. Maybe you can then remove the `poll` below.

Reply:
The problem here is: the executor service will start a separate thread and return to the main thread, which then returns to the calling code. In the tests, sometimes the method to check whether or not the file has changed is called from the main thread before the poll method in the other thread is finished, thus returning the wrong result and making the test fail. By calling `poll` explicitly in the main thread we can make sure that the first poll is finished before we return to the calling method. Given that it is done almost at time 0 after the polling thread is started, I think it makes sense to not run the background poll at the same time again, that's why I set the initial delay to the pollInterval.